### PR TITLE
Fix "bouncy" blocks

### PR DESCRIPTION
**Pull request by Miner59**
Remove some code which handles jumping on bouncy blocks. The removed code only lead to problems with jumping insanly high on blocks which belong to group "bouncy". If I remove it all works well.
I found no problems occuring when removing this code.

Fixes this issue: https://github.com/minetest/minetest/issues/1552

Sorry for my bad description, i'm not a very good english speaker.

### CODE REVIEW

**Comment by Calinou** [comment]
:+1: Tested, confirmed working.

**Comment by Miner59** [comment]
Nice, thanks for testing. Did you notice even mobs can bounce on a trampoline.
I just recognised this :-) Looks funny.

And there is another pull request which is needed to make jumping mod fully working:

https://github.com/minetest/minetest/pull/2241

This fixes the problem with players still able to jump on blocks which have disable_jump set as group.
I tested this and it worked.

**Comment by Wuzzy2** [comment]
Finally!
I have waited very long for this. The bug made bouncy blocks nearly unusable.

**Comment by paramat** [comment]
See discussion on IRC http://irc.minetest.ru/minetest-dev/2015-07-05#i_4311317 16:35 onwards

**Comment by tenplus1** [comment]
Used latest 3932 build and the bouncy blocks are still glitched when using jump while standing on them, you get shot into the air...

**Comment by Miner59** [comment]
I don't understand exactly why this pull request isn't accepted. I tested it myself and Calinou confirms it works, too.

Now I added handling of the situation, when jump key is pressed while bouncing on a bouncy block.
It gives the player a slight speed boost when he bounces back.

If you want to test it, please use only original jumping mod: https://github.com/Jeija/minetest-mod-jumping
Calinou's version has disable_jump set as a workaround for the jumping bug, so it can't work right.

**Comment by Wuzzy2** [comment]
Read what paramat has written. The chatlog indicates the code does not really fix the bug but instead introduces some problems on its own.

**Comment by Miner59** [comment]
I looked at the chatlog and I still think hmmm misunderstood something there. They tested calinou's version where you aren't allowed to jump when standing on trampoline because this was considered better than jumping insane high instead.

I can offer hmmm or other developers which want to test this PR to join my server, which has address 80.128.150.136 Port 30000.
Visit it with a minetest version with this pull request here added and I can show you exactly that you can bounce very well on trampolines now.
I will wait there for anyone the next hour.

**Comment by Calinou** [comment]
:+1:

**2015-07-13:** Tested Miner59's updates and it works as expected.

**Comment by tenplus1** [comment]
Am using latest mint/ubuntu build 3962 and jumping remains the same, holding hump button on a bouncy block shoots the player into to air and hits an unloaded ceiling...

File: src/localplayer.cpp [hunk 1]
```diff
@@ -371,6 +361,21 @@ void LocalPlayer::move(f32 dtime, Environment *env, f32 pos_max_d,
 	m_can_jump = touching_ground && !in_liquid;
 	if(itemgroup_get(f.groups, "disable_jump"))
 		m_can_jump = false;
+	// Jump key pressed while jumping off from a bouncy block
+	if (m_can_jump && control.jump && itemgroup_get(f.groups, "bouncy") &&
+		m_speed.Y >= -0.5 * BS)
+	{
+		float jumpspeed = movement_speed_jump * physics_override_jump;
+		if (m_speed.Y > 1)
```
Comment (est31) [inline src/localplayer.cpp]: { and if (...) should be on the same line, similar above.

**Comment by est31** [comment]
060e56b24c21c4cda76fcac15eecbdce9682d16a

**Pull request closed without merging**